commands/azureCommands/acr-build.ts

```diff
@@ -112,3 +107,12 @@ function getTempSourceArchivePath(): string {
     let tarFilePath = url.resolve(os.tmpdir(), `sourceArchive${id}.tar.gz`);
     return tarFilePath;
 }
+
+function filter(list: string[]): string[] {
+    for (let file of list) {
+        if (vcsIgnoreList.indexOf(file) !== -1) {
+            list.splice(list.indexOf(file), 1);
```

Comment:
I'm always cautious about updating the list in the enumeration. Can you hold the filtered item in a new list and return the new list?